compiler/AST/expr.cpp

```diff
@@ -6069,11 +6070,14 @@ new_Expr(const char* format, va_list vl) {
       const char* str = asubstr(&format[i], &format[i+n]);
       i += n-1;
       if (!strcmp(str, "TYPE")) {
-        BlockStmt* block = toBlockStmt(stack.v[stack.n-1]);
+        INT_ASSERT(stack.size() > 0);
```

Comment:
INT_FATAL allows you to add a description of the error, so is sometimes more self-documenting than INT_FATAL.  You could, for example, say
  if (stack.size == 0)
    INT_FATAL("You neglected to provide a \"{ TYPE ...\" for a block type statement.");